scrapers/MindGeekAPI.py

```diff
@@ -0,0 +1,286 @@
+                urlparse(url).netloc), file=sys.stderr)
+            _, headers = scraping_url(url)
+            # Filter the filename to remove possible mistake
+
+            search_url = 'https://site-api.project1service.com/v2/releases?title={}&type=scene'.format(
+                title_filter)
+            api_json = send_request(search_url, headers)
+            for result in api_json:
+                title_filename = ""
+                try:
+                    filename = result['videos']['mediabook']['files']["320p"]['urls']['download']
+                    title_filename = re.sub(r'^.+filename=', '', filename)
+                    title_filename = re.sub(r'_.+$', '', title_filename)
+                except:
+                    pass
+                if title_filename:
+                    making_url = re.sub(
+                        r'/\d+/*.+', '/' + str(result.get("id")) + "/" + title_filename, url)
+                else:
+                    making_url = re.sub(
+                        r'/\d+/*.+', '/' + str(result.get("id")) + "/", url)
+                save_json(result, making_url)
+                ratio = difflib.SequenceMatcher(
+                    None, title_filter, result.get('title')).ratio()
+                print("Found:{}\nRatio:{}".format(
+                    result.get('title'), ratio), file=sys.stderr)
+                if ratio > SET_RATIO:
+                    return result, making_url, headers
+        print_exit("Didn't find a match")
+    else:
+        print_exit("Can't search the scene ({} is missing)".format(
+            SET_FILE_URL))
+
+
+def send_request(url, headers):
+    try:
+        r = requests.get(url, headers=headers, timeout=(3, 5))
+    except requests.exceptions.RequestException:
+        print("An error has occurred", file=sys.stderr)
+        print("Request status: {}".format(r.status_code), file=sys.stderr)
+        try:
+            print("Message: {}".format(r.json()[0].get('message')), file=sys.stderr)
+        except:
+            pass
+        print("Check your MindGeekAPI.log for more details", file=sys.stderr)
+        with open("MindGeekAPI.log", 'w', encoding='utf-8') as f:
+            f.write("Headers used: {}\n".format(headers))
+            f.write("API URL: {}\n".format(url))
+            f.write("Response:\n{}".format(r.text))
+        sys.exit(1)
+    try:
+        api_json = r.json().get('result')
+    except:
+        print_exit("Error getting the JSON from request", file=sys.stderr)
+    return api_json
+
+# Scrape JSON for Stash
+
+
+def scraping_json(api_json, url=""):
+    scrape = {}
+    scrape['title'] = api_json.get('title')
+    date = datetime.strptime(api_json.get(
+        'dateReleased'), '%Y-%m-%dT%H:%M:%S%z')
+    scrape['date'] = str(date.date())
+    scrape['details'] = api_json.get('description')
+    if url:
+        scrape['url'] = url
+    scrape['studio'] = {}
+    scrape['studio']['name'] = api_json['collections'][0].get('name')
```

Comment:
we get an `IndexError: list index out of range ` exception here if collections  dont have entries
eg `https://www.brazzers.com/video/4030401`

Reply:
Stupid no studio scene 😡 . Good catch 👍